src/wasm/wasm-binary.cpp

```diff
@@ -1051,13 +1057,17 @@ void WasmBinaryBuilder::readFunctions() {
   if (total != functionTypes.size()) {
     throwError("invalid function section size, must equal types");
   }
+  isInFunction = true;
   for (size_t i = 0; i < total; i++) {
     if (debug) std::cerr << "read one at " << pos << std::endl;
     size_t size = getU32LEB();
```

Comment:
currFunction and func are set below at line 1094 -- makeFunction needs params and vars that are read at line 1073

Reply:
Yes, sorry if I wasn't clear before: I am saying we should replace the use of `makeFunction` at that late point with an early call to `new Function`, and then assign the fields along the way.

`makeFunction` is just a convenience utility, and in this case here, it's better overall to not use it, since it lets us set `currFunction` earlier, which simplifies things.